### PR TITLE
Update the LICENSE to be Apache 2.0.

### PR DESCRIPTION
**Pull request by jameinel**
Update all the key files with the license header, and add the
LICENSE.txt explaining this is Apache 2.0.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,39 @@
...
     packages=['ops'],
+    classifiers=[
+        'Development Status :: 4 - Beta',
+
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+
+        "License :: OSI Approved :: Apache Software License",
```
Comment (niemeyer) [inline setup.py]: Strings are being double-quotes and single-quoted back and forth.

Comment (jameinel) [reply setup.py]: good catch. I had copied them from different sources to get the right Pip metadata, I'll clean it up.

Comment (jameinel) [reply setup.py]: I also realized we don't actually test against python 3.8, so I'll pull that out for now.

Comment (chipaca) [reply setup.py]: does this mean that #142 would require changing this?

**Comment by chipaca** [comment]
if we don't want to copyright otherwise-empty files,
```sh
$ for f in $( find . \( -name .git -o -name \*.pyc -o -empty \) -prune -o -type f -print ); do
> if ! grep -q Copyright "$f"; then ls -l "$f"; fi; done
-rw-r--r-- 1 john john 49 Feb  6 09:28 ./.flake8
-rw-r--r-- 1 john john 191 Feb 17 13:10 ./.travis.yml
-rw-r--r-- 1 john john 3339 Feb 14 21:53 ./README.md
-rw-r--r-- 1 john john 14 Feb  6 09:28 ./test/charms/test_main/config.yaml
-rw-r--r-- 1 john john 556 Feb  6 09:28 ./test/charms/test_main/metadata.yaml
-rwxr-xr-x 1 john john 357 Feb  6 09:28 ./test/bin/relation-list
-rwxr-xr-x 1 john john 215 Feb  6 09:28 ./test/bin/relation-ids
-rw-r--r-- 1 john john 27 Feb  6 09:28 ./.gitignore
```

**Pull request merged**